.github/workflows/main.yml

```diff
@@ -1082,6 +1082,16 @@ jobs:
     - run: |
         rustup component add rust-src
         rustup target add ${{ matrix.target }}
+    # Update binutils if MinGW due to https://github.com/rust-lang/rust/issues/112368
+    - run: C:/msys64/usr/bin/pacman.exe -S --needed mingw-w64-x86_64-gcc --noconfirm
+      if: matrix.target == 'x86_64-pc-windows-gnu'
+    - shell: pwsh
+      run: |
+        echo "C:\msys64\mingw64\bin" >> $Env:GITHUB_PATH
+        echo "CC=gcc" >> $Env:GITHUB_ENV
+        echo "CXX=g++" >> $Env:GITHUB_ENV
+      if: matrix.target == 'x86_64-pc-windows-gnu'
```

Comment:
Mind leaving a comment as to why these steps are necessary? (this doesn't seem to directly follow from rust-lang/rust#112368 by my read